src/olympia/addons/models.py

```diff
@@ -1910,17 +1905,28 @@ class Meta:
 class AddonCategory(models.Model):
     id = PositiveAutoField(primary_key=True)
     addon = models.ForeignKey(Addon, on_delete=models.CASCADE)
-    category = models.ForeignKey('Category', on_delete=models.CASCADE)
+    category_id = models.SmallIntegerField()
```

Comment:
Thinking: if we change this for whatever size integer is used for foreign keys in mysql we could save the database field size conversion in the migration.

Reply:
Yeah. I was thinking of doing something like `category = models.IntegerField(db_column='category_id')` but it doesn't seem to work...

Reply:
Another trick I was thinking about would be to do keep `ForeignKey` with `db_constraint=False`... And then change it to an `IntegerField` later, but the intermediary step would help convince migrations that it's ok. (Haven't completely thought this through yet)

Reply:
One final option: use `RunSQL`, providing `state_operations` to help migrations system understand what we're doing: https://docs.djangoproject.com/en/2.2/ref/migration-operations/#runsql ; then we'd be free to do the `ALTER` we want and leave things in a state migrations would understand.

Reply:
https://github.com/mozilla/addons-server/pull/16343/commits/c35b82fbac587e09b33df9696a621f40213e7c1a results in just one `ALTER` to drop the fk constraint.  It's difficult to know if that's enough, but I played around locally with tests from the master branch running on a post-migration database and they passed (addons/tests/test_models.py anyway - I didn't run the entire suite)

The drop model for the `Category` model will have to be in a later tag, obvs.

Reply:
(The field type was changed to PositiveIntegerField too so the column type doesn't have to change)